BGL/include/CGAL/boost/graph/properties.h

```diff
@@ -86,13 +91,37 @@ using boost::face_external_index;
 namespace CGAL{
 namespace helpers {
 
+//check that an existing map is initialized/valid
+template <class SimplexRange, class IndexMap>
+bool is_index_map_valid( std::size_t num_simplices,
+                        const SimplexRange& range,
+                        const IndexMap& idmap)
+{
+  
+  typedef typename boost::property_traits<IndexMap>::value_type Id_type;
+  Id_type max_id = static_cast<Id_type>(num_simplices);
+  std::vector<bool> indices(max_id + 1);
+  for(const auto& simplex : range)
+  {
+    Id_type id = get(idmap, simplex);
+    if( id >= 0 && id <= max_id && ! indices[id])
+      indices[id] = true;
+    else
+      return false;
+  }
+  return true;
+}
+
+
 // matches read-write property maps
 template <class PolygonMesh, class FaceIndexMap, class Tag>
 void init_face_indices(PolygonMesh& pm,
                        FaceIndexMap& fid,
                        boost::read_write_property_map_tag,
                        Tag)
 {
+  if(is_index_map_valid(num_faces(pm), faces(pm), fid))
```

Comment:
`num_faces(pm)` is not necessarily equal to the size of `faces(pm)`

Reply:
?

Reply:
You're checking if the faces are well indexed by defining it as a unique ID between 0 and num_faces(pm), but well indexed should likely be defined as each face has a unique ID between 0 and faces(pm).size(). Not sure if it'd actually mess up some algorithms, but it just seems safer.

Reply:
That's not true, the code is correct and it should be between 0 and num_faces-1 (think about defaut index map for a surface mesh with garbage).

Reply:
I understand that it looks correct from the data structure POV, but looking from the algorithm POV, don't algorithms expect the face IDs to respresent a unique ID between 0 and the size of the range of faces being passed? For example if I make a vector `vector<whatever> v(faces(pm).size())` and decide to do `v[get(fidmap, f)] = my_whatever;` then it'll explode. Sure I should have known better than the size of the range is smaller than `num_faces(pm)`, but it's creating an opening for user failure.

On the other side, I imagine that in cases where the fidmap is based on just the index of the element in an index-based data structures, then it would be annoying to have to recreate a "proper" ID map.

Reply:
This does come from boost and it actually make sense if you have an algorithm removing elements (you do not want to corrupt properties of element that were not touched). If you need such a strict index, we could have a wrapper with an extra indirection through a vector.